includes/class-amp-reader-themes.php

```diff
@@ -0,0 +1,512 @@
+				'avatar'        => 'https://secure.gravatar.com/avatar/61ee2579b8905e62b4b4045bdc92c11a?s=96&d=monsterid&r=g',
+				'display_name'  => 'WordPress.org',
+			],
+			'screenshot_url' => '//ts.w.org/wp-content/themes/twentyeleven/screenshot.png?ver=3.4',
+			'rating'         => 94,
+			'num_ratings'    => '45',
+			'homepage'       => 'https://wordpress.org/themes/twentyeleven/',
+			'description'    => 'The 2011 theme for WordPress is sophisticated, lightweight, and adaptable. Make it yours with a custom menu, header image, and background -- then go further with available theme options for light or dark color scheme, custom link colors, and three layout choices. Twenty Eleven comes equipped with a Showcase page template that transforms your front page into a showcase to show off your best content, widget support galore (sidebar, three footer areas, and a Showcase page widget area), and a custom "Ephemera" widget to display your Aside, Link, Quote, or Status posts. Included are styles for print and for the admin editor, support for featured images (as custom header images on posts and pages and as large images on featured "sticky" posts), and special styles for six different post formats.',
+			'requires'       => false,
+			'requires_php'   => '5.2.4',
+		],
+	];
+
+	/**
+	 * Retrieves all AMP plugin options specified in the endpoint schema.
+	 *
+	 * @return array Formatted theme data.
+	 */
+	public function get_themes() {
+		if ( null !== $this->themes ) {
+			return $this->themes;
+		}
+
+		$themes   = $this->get_default_supported_reader_themes();
+		$themes   = array_map( [ $this, 'prepare_theme' ], $themes );
+		$themes[] = $this->get_classic_mode();
+
+		/**
+		 * Filters supported reader themes.
+		 *
+		 * @param array $themes [
+		 *     Reader theme data.
+		 *     {
+		 *         @type string         $name           Theme name.
+		 *         @type string         $slug           Theme slug.
+		 *         @type string         $slug           URL of theme preview.
+		 *         @type string         $screenshot_url The URL of a mobile screenshot. Note: if this is empty, the theme may not display.
+		 *         @type string         $homepage       A link to a page with more information about the theme.
+		 *         @type string         $description    A description of the theme.
+		 *         @type string|boolean $requires       Minimum version of WordPress required by the theme. False if all versions are supported.
+		 *         @type string|boolean $requires_php   Minimum version of PHP required by the theme. False if all versions are supported.
+		 *         @type string         $download_link  A link to the theme's zip file. If empty, the plugin will attempt to download the theme from wordpress.org.
+		 *     }
+		 * ]
+		 */
+		$themes = array_filter( (array) apply_filters( 'amp_reader_themes', $themes ) );
+
+		foreach ( $themes as &$theme ) {
+			$theme['availability'] = $this->get_theme_availability( $theme );
+		}
```

Comment:
Maybe marginally better to use `array_map()` here, as passing variables by reference can be a headache.
```suggestion
		$themes = array_map(
			static function ( $theme ) {
				$theme['availability'] = $this->get_theme_availability( $theme );
				return $theme;
			},
			$themes
		);
```